Userland/Applications/Browser/Tab.cpp

```diff
@@ -38,8 +39,14 @@
 
 namespace Browser {
 
+static String s_search_engine_format = {};
+
 URL url_from_user_input(const String& input)
 {
+    if (input.starts_with("?") && !s_search_engine_format.is_null()) {
+        return URL(String::formatted(s_search_engine_format, input.substring(1)));
```

Comment:
I think it might be necessary to URL-encode the input as well, so it doesn't allow injection of additional query parameters etc.

```suggestion
        return URL(String::formatted(s_search_engine_format, urlencode(input.substring(1))));
```

Reply:
Good idea :^)